File: c/prime_number.c
#include <stdio.h>

void main()
{
    int num = 0;
    int is_prime = 1;
    printf("Digite um número: ");
    scanf("%d", &num);
    for (int count = num - 1; count > 1; count--)
    {
        if (num % count == 0) {
            is_prime = 0;
            break;
        }
    }
    
    if (is_prime && num != 1) {
        printf("O número é primo.");
    } else {
        printf("O número não é primo.");
    }
    return 0;
}

File: go/prime_number.go
package main

import (
	"fmt"
	"math"
)

func main() {
    var x int
    var is_primo bool = true
    fmt.Print("Digite um valor: ")
    fmt.Scan(&x)
    for i := 2.0; i <= math.Floor(float64 (x) / 2); i++{
        if math.Mod(float64 (x), i) == 0 {
            is_primo = false
            break
        }
    }
    if x == 1{
        is_primo = false
    }
    if is_primo {
        fmt.Printf("O número %v é primo.", x)
    } else {
        fmt.Printf("O número %v não é primo.", x)
    }

}